src/main/java/net/oauth/jsontoken/JsonToken.java

```diff
@@ -82,11 +79,17 @@ public JsonToken(Signer signer) {
   public JsonToken(Signer signer, Clock clock) {
     Preconditions.checkNotNull(signer);
     Preconditions.checkNotNull(clock);
-    
+
+    header = new JsonObject();
+    header.addProperty(ALGORITHM_HEADER, signer.getSignatureAlgorithm().getNameForJson());
```

Comment:
Can we do a null check here for signer.getSignatureAlgorithm()? Even though it's a required field according to the JWT spec, previously this would only throw an IllegalStateException later when the header is actually needed somewhere (e.g. when signing the jwt) rather than throwing NPE in the constructor.